crates/ruff_linter/src/rules/pydoclint/snapshots/ruff_linter__rules__pydoclint__tests__docstring-extraneous-parameter_DOC102_google.py.snap

```diff
@@ -0,0 +1,182 @@
+---
+source: crates/ruff_linter/src/rules/pydoclint/mod.rs
+---
+DOC102_google.py:6:1: DOC102 `a` is not in the function's signature
+   |
+ 4 |       Adds two numbers and returns the result.
+ 5 |   
+ 6 | /     Args:
+ 7 | |         a (int): The first number to add.
+ 8 | |         b (int): The second number to add.
+ 9 | | 
+10 | |     Returns:
+   | |_^ DOC102
+11 |           int: The sum of the two numbers.
+12 |       """
+   |
+   = help: Remove the extraneous parameter from the docstring
+
+DOC102_google.py:21:1: DOC102 `multiplier` is not in the function's signature
+   |
+19 |       Multiplies each element in a list by a given multiplier.
+20 |   
+21 | /     Args:
+22 | |         lst (list of int): A list of integers.
+23 | |         multiplier (int): The multiplier for each element in the list.
```

Comment:
Highlighting a multiline-range always leads to somewhat awkward code frame rendering. I think it would be better to change the diagnostic range to the `multiplier` code frame rather than the entire args section. Or is there precedence in other pydocstyle rules to highlight the entire section?

If this range is due to creating a single diagnostic for all the function's extraneous parameters than I think I prefer having multiple diagnostics instead (also makes it easier to suppress an error in case that's needed without risking silencing new extraneous parameters)

Reply:
That is how I have done it previously for example DOC502. The main decision factor is that it's a bit awkward to get the range of the single entry.